diff --git a/waftools/detections/compiler_swift.py b/waftools/detections/compiler_swift.py
index d4ba162667..6191fd8ec2 100644
--- a/waftools/detections/compiler_swift.py
+++ b/waftools/detections/compiler_swift.py
@@ -29,13 +29,13 @@ def __add_swift_flags(ctx):
 def __add_swift_library_linking_flags(ctx, swift_library):
     ctx.env.append_value('LINKFLAGS', [
         '-L%s' % swift_library,
-        '-Xlinker', '-force_load_swift_libs', '-lc++',
+        '-lc++',
     ])
 
 def __find_swift_library(ctx):
     swift_library_paths = [
-        'Toolchains/XcodeDefault.xctoolchain/usr/lib/swift_static/macosx',
-        'usr/lib/swift_static/macosx'
+        'Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx',
+        'usr/lib/swift/macosx'
     ]
     dev_path = __run(['xcode-select', '-p'])[1:]
 
